kubernetes/kubernetes.go

```diff
@@ -229,25 +233,26 @@ func CollectKubeMetrics(config KubeAgentConfig) {
 // isCustomUploadEnvsSet checks the KubeAgentConfig for custom upload ENVs and returns true/false and the provider mode
 // Currently supported: AWS and Azure, the goal is to add more support with this function later
 func isCustomUploadEnvsSet(ka *KubeAgentConfig) (bool, string) {
-	var isCustomUpload bool
-	var agentMode string
-	isCustomUpload, agentMode = isCustomS3UploadEnvsSet(ka)
-	if isCustomUpload {
-		return isCustomUpload, agentMode
-	}
-	isCustomUpload, agentMode = isCustomAzureUploadEnvsSet(ka)
-	return isCustomUpload, agentMode
+	isCustomS3Upload := isCustomS3UploadEnvsSet(ka)
+	if isCustomS3Upload {
+		return isCustomS3Upload, awsProvider
+	}
+	isCustomBlobUpload := isCustomAzureUploadEnvsSet(ka)
+	if isCustomBlobUpload {
+		return isCustomBlobUpload, azureProvider
+	}
+	return false, ""
 }
 
 // isCustomS3UploadEnvsSet checks to see if the agent has a custom S3 location and S3 region to upload to
 // if both these variables are not set, default upload to Apptio S3
-func isCustomS3UploadEnvsSet(ka *KubeAgentConfig) (bool, string) {
+func isCustomS3UploadEnvsSet(ka *KubeAgentConfig) bool {
 	if ka.CustomS3Region == "" && ka.CustomS3UploadBucket == "" {
 		if ka.APIKey == "" {
 			log.Fatalf("Invalid agent configuration. CLOUDABILITY_API_KEY is required " +
```

Comment:
Have you tested the agent in the azure configuration? Seems like this would just crash.

I don't think a bunch of log.Fatal's is a great handling method for this when multiple configurations need to be interpreted.

Reply:
Yes tested the agent with the azure configs. When the aws config isn't set both ka.CustomS3Region and ka.CustomS3UploadBucket are set to "" the API key must be set so it will return false. Then it will check the azure config.

The log.Fatal's seem to be necessary to break the process when a custom upload config is partially set but not all values are there so it will not revert to the default upload. Also the log will show the customer which env config is not set correctly.